src/Tools/Source/RunTests/Cache/ContentUtil.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RunTests.Cache
+{
+    internal sealed class ContentUtil
+    {
+        private readonly Options _options;
+        private readonly MD5 _hash = MD5.Create();
+        private readonly Dictionary<string, string> _fileToChecksumMap = new Dictionary<string, string>();
+
+        internal ContentUtil(Options options)
+        {
+            _options = options;
+        }
+
+        internal ContentFile GetTestResultConentFile(string assemblyPath)
+        {
+            try
+            {
+                var content = BuildTestResultContent(assemblyPath);
+                var checksum = GetChecksum(content);
+                return new ContentFile(checksum: checksum, content: content);
+            }
+            catch (Exception ex)
+            {
+                Logger.Log($"Error creating log file {ex.Message} {ex.StackTrace}");
+                return ContentFile.Empty;
+            }
+        }
+
+        private string BuildTestResultContent(string assemblyPath)
+        {
+            var builder = new StringBuilder();
+            builder.AppendLine($"Assembly: {Path.GetFileName(assemblyPath)} {GetFileChecksum(assemblyPath)}");
+
+            var configFilePath = $"{assemblyPath}.config";
+            var configFileChecksum = File.Exists(configFilePath)
+                ? GetFileChecksum(configFilePath)
+                : "<no config file>";
+            builder.AppendLine($"Config: {Path.GetFileName(configFilePath)} {configFileChecksum}");
+
+            builder.AppendLine($"Xunit: {Path.GetFileName(_options.XunitPath)} {GetFileChecksum(_options.XunitPath)}");
+            AppendReferences(builder, assemblyPath);
+            builder.AppendLine("Options:");
+            builder.AppendLine($"\t{nameof(_options.Test64)} - {_options.Test64}");
+            builder.AppendLine($"\t{nameof(_options.UseHtml)} - {_options.UseHtml}");
+            builder.AppendLine($"\t{nameof(_options.Trait)} - {_options.Trait}");
+            builder.AppendLine($"\t{nameof(_options.NoTrait)} - {_options.NoTrait}");
+
+            // TODO: Need to include dependency information here, option data, etc ...
+            // Test file alone isn't enough.  Makes it easy to test though.
+            return builder.ToString();
+        }
+
+        private void AppendReferences(StringBuilder builder, string assemblyPath)
+        {
+            builder.AppendLine("References:");
+
+            var initialAssembly = Assembly.ReflectionOnlyLoadFrom(assemblyPath);
```

Comment:
It can easily be done for assemblies for which I have a file path.  For GAC assemblies though, when I use the `ReflectionOnlyLoad` method below, I don't have a path.  In that case MetadataReader wasn't an option that I knew how to work with . Hence I went with this route.